### PR TITLE
Typescript support

### PR DESCRIPTION
**Pull request by bjornkpu**
# Add Typescript scaffold.

- Add dependencies
- Add tsconfig that extends recommended config
- Add eslint config
- Update .gitignore to ignore js files in ts-dir
- Add README
- Add jest config for testing

# Add TS source files.

## app
- Add server that lifts the server
- Add app that defines routes. Separates this to make testing easier

## workplace
- Add request interface and enums
- Add route for webhook server verification
- Add route for webhook post
- Add handler for group entries. currently just prints to console

## Test
- Add happy-path tests for workplace routes. Mainly to run the endpoints without having to post in workplace

### CODE REVIEW

File: node-ts/src/routes/workplace.routes.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    
+    if (!(mode === 'subscribe' && verify_token === WORKPLACE_VERIFY_TOKEN)) {
+        log.error('Failed validation. Make sure the validation tokens match.')
+        res.sendStatus(403);
+        return
+    }
+    
+    log.debug('Validating webhook')
+    res.status(200).send(challenge)
```
[the remark this answers is not in the document]
Comment (bjornkpu) [reply node-ts/src/routes/workplace.routes.ts:29]: That is the whole point with the challenge. and with the verifyRequestSignature() check we can be sure the requests comes from one of FaceBooks servers

[the remark this answers is not in the document]
Comment (tplive) [reply node-ts/src/routes/workplace.routes.ts:29]: Should probably heed this warning... :)

[the remark this answers is not in the document]
Comment (tplive) [reply node-ts/src/routes/workplace.routes.ts:29]: Yes, but the `challenge` should be escaped. :)

[the remark this answers is not in the document]
Comment (bjornkpu) [reply node-ts/src/routes/workplace.routes.ts]: Fixed in [d0566ac](https://github.com/evry-ace/iskaribot/pull/58/commits/d0566acb3c8f6b44ee9c6a0f6bee59ba7878ceeb)

File: node-ts/src/routes/workplace.routes.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        log.error(e);
+    } finally {
+        res.sendStatus(200);
+    }
+});
+
+/** Verify that the callback came from Facebook. */
+function verifyRequestSignature(payload:unknown,signature:unknown) {
+    if (!signature) throw new Error('No signature in header.')
+    // const signature = req.headers['x-hub-signature'];
```
Comment (tplive) [inline node-ts/src/routes/workplace.routes.ts:54]: Keeping in mind that this is still a work in progress, we should still avoid commented-out code. :)

**Review by tplive — APPROVED** [review]
LGTM! :)

**Pull request merged**